edge-modules/MessagesAnalyzer/Settings.cs

```diff
@@ -32,21 +32,21 @@ class Settings
 
                 return new Settings(
                     configuration.GetValue<string>(EventHubConnectionStringPropertyValue),
-                    configuration.GetValue(ConsumerGroupIdPropertyName, DefaultConsumerGroupId),
-                    configuration.GetValue(DeviceIdPropertyName, DefaultDeviceId),
+                    configuration.GetValue<string>(ConsumerGroupIdPropertyName, DefaultConsumerGroupId),
+                    configuration.GetValue<string>(DeviceIdPropertyName),
                     excludedModules,
-                    configuration.GetValue(WebhostPortPropertyName, DefaultWebhostPort),
-                    configuration.GetValue(ToleranceInMillisecondsPropertyName, DefaultToleranceInMilliseconds));
+                    configuration.GetValue<string>(WebhostPortPropertyName, DefaultWebhostPort),
```

Comment:
legacy issue. And even better if we validate it between 0 and 65535.  Btw I approved this PR, your call if you want to fix it this PR or not.
```suggestion
                    configuration.GetValue<int>(WebhostPortPropertyName, DefaultWebhostPort),
```